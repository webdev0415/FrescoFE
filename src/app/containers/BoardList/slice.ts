import { PayloadAction } from '@reduxjs/toolkit';
import { createSlice } from 'utils/@reduxjs/toolkit';
import { ContainerState } from './types';

// The initial state of the BoardList container
export const initialState: ContainerState = {
  loading: false,
  boardList: [],
};

const boardListSlice = createSlice({
  name: 'boardList',
  initialState,
  reducers: {
    attemptGetBoards(state, action) {
      state.loading = true;
    },
    getBoardsSuccess(state, action: PayloadAction<any>) {
      state.boardList = action.payload;
      state.loading = false;
    },
    getBoardsFail(state) {
      state.boardList = [];
      state.loading = false;
    },
    deleteBoard(state, action) {
      console.log('action', action);
      console.log('state.boardList', state.boardList);
      state.boardList = state.boardList.filter(
        board => board.id !== action.payload,
      );
    },
  },
});

export const { actions, reducer, name: sliceKey } = boardListSlice;
