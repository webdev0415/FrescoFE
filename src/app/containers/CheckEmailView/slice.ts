import { PayloadAction } from '@reduxjs/toolkit';
import { createSlice } from 'utils/@reduxjs/toolkit';
import { ContainerState } from './types';

// The initial state of the CheckEmailView container
export const initialState: ContainerState = {};

const checkEmailViewSlice = createSlice({
  name: 'checkEmailView',
  initialState,
  reducers: {
    someAction(state, action: PayloadAction<any>) {},
  },
});

export const { actions, reducer, name: sliceKey } = checkEmailViewSlice;
