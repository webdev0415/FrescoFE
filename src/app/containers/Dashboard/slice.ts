import { PayloadAction } from '@reduxjs/toolkit';
import { createSlice } from 'utils/@reduxjs/toolkit';
import { ContainerState } from './types';

// The initial state of the Dashboard container
export const initialState: ContainerState = {
  loading: false,
  listEmail: [],
  linkInvitation: '',
};

const dashboardSlice = createSlice({
  name: 'dashboard',
  initialState,
  reducers: {
    searchEmailRequest(state, action: PayloadAction<any>) {
      state.linkInvitation = '';
      // state.loading = true;
    },

    // TODO: define Payload type and use here
    searchEmailSuccess(state, action: PayloadAction<any>) {
      // console.log(action.payload);
      state.listEmail = action.payload;
      // state.loading = false;
    },

    searchEmailError(state, action: PayloadAction<any>) {
      // state.loading = false;
    },

    invitationRequest(state, action: PayloadAction<any>) {
      state.linkInvitation = '';
      state.loading = true;
    },

    // TODO: define Payload type and use here
    invitationSuccess(state, action: PayloadAction<any>) {
      state.linkInvitation = action.payload;
      state.loading = false;
    },

    invitationError(state, action: PayloadAction<any>) {
      state.loading = false;
    },
  },
});

export const { actions, reducer, name: sliceKey } = dashboardSlice;
