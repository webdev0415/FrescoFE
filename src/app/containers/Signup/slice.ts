import { PayloadAction } from '@reduxjs/toolkit';
import { createSlice } from 'utils/@reduxjs/toolkit';
import { ContainerState } from './types';

// The initial state of the Signup container
export const initialState: ContainerState = {
  token: null,
  loading: false,
  errorCode: null,
};

const signupSlice = createSlice({
  name: 'signup',
  initialState,
  reducers: {
    signUpRequest(state, action: PayloadAction<any>) {
      state.errorCode = null;
      state.loading = true;
    },

    // TODO: define Payload type and use here
    signUpSuccess(state) {
      state.loading = false;
    },

    signUpError(state, action: PayloadAction<any>) {
      state.errorCode = action.payload.status;
      state.loading = false;
    },

    signUpErrorReset(state) {
      state.errorCode = null;
      state.loading = false;
    },
  },
});

export const { actions, reducer, name: sliceKey } = signupSlice;
