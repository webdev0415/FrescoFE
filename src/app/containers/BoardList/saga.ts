import { all, put, takeEvery } from 'redux-saga/effects';
import { message } from 'antd';
import { actions } from './slice';
import axios from 'axios';
import Auth from 'services/Auth';
axios.defaults.headers.common['Authorization'] = `Bearer ${Auth.getToken()}`;

// import { actions as globalActions } from '../../slice';
export function* getBoards(action) {
  const { payload } = action;

  try {
    axios.defaults.headers.common[
      'Authorization'
    ] = `Bearer ${Auth.getToken()}`;
    const response = yield axios.get(`/board/organization/${payload}`);

    yield put(actions.getBoardsSuccess(response.data));
  } catch (error) {
    yield put(actions.getBoardsFail());
    message.error(error.message);
  }
}

export function* boardListSaga() {
  yield all([takeEvery(actions.attemptGetBoards.type, getBoards)]);
}
