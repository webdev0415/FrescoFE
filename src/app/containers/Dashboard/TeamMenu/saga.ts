import { all, put, takeLatest } from 'redux-saga/effects';
import { message } from 'antd';
import axios from 'axios';
import { actions } from './slice';

export function* teamMenu(action) {
  const { payload } = action;
  const { orgId, token } = payload;
  try {
    axios.defaults.headers.common['Authorization'] = `Bearer ${token}`;
    const response = yield axios.get('teams?orgId=' + orgId.orgId);
    yield put(actions.getTeamMenuSuccess(response.data));
  } catch (error) {
    yield put(
      actions.getTeamMenuError({
        message: error.message,
        status: error.response.status,
      }),
    );
    message.error(error.message);
  }
}

export function* teamMenuSaga() {
  yield all([takeLatest(actions.getTeamMenuRequest.type, teamMenu)]);
}
