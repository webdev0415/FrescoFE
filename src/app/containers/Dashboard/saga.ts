import { all, put, takeLatest } from 'redux-saga/effects';
import { message } from 'antd';
import axios from 'axios';
import { actions } from './slice';
import { v4 as uuidv4 } from 'uuid';

// import { actions as globalActions } from '../../slice';

export function* searchEmail(action) {
  const { payload } = action;
  // console.log('payload', payload);
  try {
    axios.defaults.headers.common['Authorization'] = `Bearer ${payload.token}`;
    const response = yield axios.post('/users/suggest', {
      email: payload.data.email,
      orgId: payload.data.orgId,
    });
    // console.log('response', response);
    yield put(actions.searchEmailSuccess(response.data));
  } catch (error) {
    yield put(
      actions.searchEmailError({
        message: error.message,
        status: error.response.status,
      }),
    );
    message.error(error.message);
  }
}

export function* invitation(action) {
  const { payload } = action;
  const { history, permission, token, orgId, data } = payload;
  const feToken = uuidv4();
  // console.log('payload', data, payload?.data[0], feToken);
  try {
    axios.defaults.headers.common['Authorization'] = `Bearer ${token}`;
    const response = yield axios.post('invitation', {
      toUserId: data[0]?.id || '',
      toEmail: data[0]?.email || data.email,
      token: feToken,
      permission,
      orgId,
    });
    // console.log('response', response);
    yield put(actions.invitationSuccess(response.data));
    message.success('Invitation successfully.');
  } catch (error) {
    history.push('/err');
    yield put(
      actions.invitationError({
        message: error.message,
        status: error.response.status,
      }),
    );
    message.error(error.message);
  }
}

export function* dashboardSaga() {
  yield all([
    takeLatest(actions.searchEmailRequest.type, searchEmail),
    takeLatest(actions.invitationRequest.type, invitation),
  ]);
}
