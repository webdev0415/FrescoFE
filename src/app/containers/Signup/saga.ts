import { all, put, takeLatest } from 'redux-saga/effects';
import { message, Modal } from 'antd';
import axios from 'axios';

import { actions } from './slice';
function sleep(sec) {
  return new Promise(resolve => setTimeout(resolve, sec * 1000));
}
export function* signUp(action) {
  try {
    const { payload } = action;
    const { data, history } = payload;
    yield axios.post('auth/register', data);
    yield put(actions.signUpSuccess());
    message.success(`You're registered successfully.`);
    history.push('/auth/check-email');
  } catch (error) {
    if (error.response.status !== 409) {
      message.error(error.message);
    }
    yield put(actions.signUpError(error.response));
    yield sleep(2);
    yield put(actions.signUpErrorReset());
  }
}

export function* signupSaga() {
  yield all([takeLatest(actions.signUpRequest.type, signUp)]);
}
