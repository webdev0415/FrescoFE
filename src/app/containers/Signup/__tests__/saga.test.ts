import { put } from 'redux-saga/effects';
import { actions } from '../slice';

import { signUp, signupSaga } from '../saga';

describe('SignUp-> Saga-> signUp', () => {
  let signUpSagaIterator: ReturnType<typeof signUp>;
  beforeEach(() => {
    const data = {
      username: 'example',
      workEmail: 'example@fres',
      password: 'admin123',
    };

    const action = actions.signUpRequest({
      history: { push: () => {} },
      data,
    });

    signUpSagaIterator = signUp(action);

    const { payload } = action;

    const requestDescriptor = signUpSagaIterator.next(payload).value;

    expect(requestDescriptor).toMatchSnapshot();
  });

  it('signUp success', () => {
    const successData = {
      username: 'example',
      role: 'USER',
    };
    const response = { data: successData };
    const postDescriptorSignUpSuccess = signUpSagaIterator.next(response).value;

    expect(postDescriptorSignUpSuccess).toEqual(put(actions.signUpSuccess()));

    const iteration = signUpSagaIterator.next();
    expect(iteration.done).toBe(true);
  });

  it('signUp error 409', () => {
    const postDescriptorSignUpError = signUpSagaIterator.throw({
      response: { status: 409 },
    }).value;
    expect(postDescriptorSignUpError).toEqual(
      put(actions.signUpError({ status: 409 })),
    );
    signUpSagaIterator.next();
    signUpSagaIterator.next();
    signUpSagaIterator.next();
    const iteration = signUpSagaIterator.next();
    expect(iteration.done).toBe(true);
  });
  it('signUp error 500', () => {
    const postDescriptorSignUpError = signUpSagaIterator.throw({
      response: { status: 500 },
    }).value;
    expect(postDescriptorSignUpError).toEqual(
      put(actions.signUpError({ status: 500 })),
    );
    signUpSagaIterator.next();
    signUpSagaIterator.next();
    signUpSagaIterator.next();
    const iteration = signUpSagaIterator.next();
    expect(iteration.done).toBe(true);
  });
  it('signUp error reset', () => {
    const postDescriptorSignUpError = signUpSagaIterator.throw({
      response: { status: 409 },
    }).value;
    signUpSagaIterator.next();
    const postDescriptorSignUpErrorReset = signUpSagaIterator.next().value;
    expect(postDescriptorSignUpErrorReset).toEqual(
      put(actions.signUpErrorReset()),
    );

    const iteration = signUpSagaIterator.next();
    expect(iteration.done).toBe(true);
  });
});

describe('SignUp-> Saga-> signUpSaga', () => {
  it('signUpSaga watch all', () => {
    let signUpSagaWatchAllIterator: ReturnType<typeof signupSaga>;
    signUpSagaWatchAllIterator = signupSaga();
    const watchDescriptor = signUpSagaWatchAllIterator.next().value;
    expect(watchDescriptor).toMatchSnapshot();
  });
});
