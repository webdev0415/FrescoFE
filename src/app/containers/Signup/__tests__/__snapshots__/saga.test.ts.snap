// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUp-> Saga-> signUp signUp error 409 1`] = `Promise {}`;

exports[`SignUp-> Saga-> signUp signUp error 500 1`] = `Promise {}`;

exports[`SignUp-> Saga-> signUp signUp error reset 1`] = `Promise {}`;

exports[`SignUp-> Saga-> signUp signUp success 1`] = `Promise {}`;

exports[`SignUp-> Saga-> signUpSaga signUpSaga watch all 1`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": true,
  "payload": Array [
    Object {
      "@@redux-saga/IO": true,
      "combinator": false,
      "payload": Object {
        "args": Array [
          "signup/signUpRequest",
          [Function],
        ],
        "context": null,
        "fn": [Function],
      },
      "type": "FORK",
    },
  ],
  "type": "ALL",
}
`;
